### PR TITLE
Remove M mode component from adversary.

### PR DESCRIPTION
**Pull request by jyao1**
Fix https://github.com/riscv-non-isa/riscv-ap-tee-io/issues/77

### CODE REVIEW

File: specification/05-security_model.adoc [hunk 1]
```diff
@@ -83,7 +83,7 @@ The CoVE-IO security model aims at protecting the above-described TVM assets
 from the following adversaries:
 
 * _Privileged host software adversary_: This includes host software components
-  executing in S and M mode like the host firmware, kernel, hypervisor, VMM,
+  executing in S/HS mode like the host firmware, kernel, hypervisor, VMM,
```
Comment (sameo) [inline specification/05-security_model.adoc:86]: Why are we removing M? Also host firmware may be an M component.

Comment (jyao1) [reply specification/05-security_model.adoc:87]: M mode component is in TCB, right?
If so, it cannot be Privileged adversary.

Comment (sameo) [reply specification/05-security_model.adoc:87]: Not all M mode components are in the confidential computing TCB.

Comment (jyao1) [reply specification/05-security_model.adoc:87]: Got you.

I am checking https://github.com/riscv-non-isa/riscv-ap-tee/blob/main/specification/threatmodel.adoc.

To align the definition, I believe "M mode component" should be "**Startup Software adversary**".

Thought?

Comment (sameo) [reply specification/05-security_model.adoc:87]: That makes perfect sense to me. We should align with the CoVE spec.

Comment (jyao1) [reply specification/05-security_model.adoc]: Move the M-mode to Startup Adversary.

**Review by sameo — APPROVED** [review]

**Pull request merged**